vela-templates/definitions/internal/trait/resource.cue

```diff
@@ -8,33 +8,40 @@ resource: {
 	}
 }
 template: {
-	patch: spec: template: spec: containers: [...{
-		resources: {
-			if parameter.cpu != _|_ && parameter.memory != _|_ && parameter.requests == _|_ && parameter.limits == _|_ {
-				requests: {
-					cpu:    parameter.cpu
-					memory: parameter.memory
+	patch: spec: template: spec: {
+		// +patchKey=name
+		containers: [{
+			resources: {
+				if parameter.cpu != _|_ && parameter.memory != _|_ && parameter.requests == _|_ && parameter.limits == _|_ {
```

Comment:
```suggestion
				if parameter.cpu != _|_ if parameter.memory != _|_ if parameter.requests == _|_ if parameter.limits == _|_ {
```

It's better to use multiple if since there's no short circuit in cue, ref: https://github.com/cue-lang/cue/issues/2232

Reply:
updated